rust-runtime/aws-smithy-runtime/src/client/interceptor.rs

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_runtime_api::client::interceptors::context::phase::BeforeTransmit;
+use aws_smithy_runtime_api::client::interceptors::{BoxError, Interceptor, InterceptorContext};
+use aws_smithy_runtime_api::config_bag::ConfigBag;
+use std::fmt;
+use std::marker::PhantomData;
+
+pub struct RequestMapInterceptor<F, E> {
```

Comment:
`MapRequestInterceptor` may be a nicer name since it has the verb in front.

Reply:
Renamed in be5a126.